apps/server/src/modules/group/repo/group.repo.ts

```diff
@@ -98,4 +157,67 @@ export class GroupRepo extends BaseDomainObjectRepo<Group, GroupEntity> {
 
 		return domainObjects;
 	}
+
+	private async findAvailableGroup(
+		pipelineStage: unknown[],
+		skip = 0,
+		limit?: number,
+		nameQuery?: string
+	): Promise<Page<Group>> {
+		let nameRegexFilter = {};
+		const pipeline: unknown[] = pipelineStage;
+
+		if (nameQuery && StringValidator.isNotEmptyString(nameQuery, true)) {
+			const escapedName = nameQuery.replace(MongoPatterns.REGEX_MONGO_LANGUAGE_PATTERN_WHITELIST, '').trim();
+			if (StringValidator.isNotEmptyString(escapedName, true)) {
+				nameRegexFilter = { name: { $regex: nameQuery, $options: 'i' } };
```

Comment:
use `escapedName `. Very important! otherwise you have a vulnerability for a ReDoS attack. check this everwhere.